kbs/src/resource/plugin/nebula.rs

```diff
@@ -0,0 +1,469 @@
+// SPDX-License-Identifier: Apache-2.0
+//
+// Copyright (c) 2024 by IBM Inc.
+
+//! The nebula plugin allows the KBS to deliver resources required to create
+//! an encrypted overlay network between nodes using [Nebula](https://github.com/slackhq/nebula),
+//!
+//! Within the Nebula overlay network, all communications between nodes are
+//! automatically encrypted.
+
+use anyhow::{anyhow, bail, Context, Result};
+use serde_qs;
+use std::ffi::OsString;
+use std::fmt;
+use std::fs;
+use std::io::Read;
+use std::path::{Path, PathBuf};
+use std::process::Command;
+use std::sync::Arc;
+use tempfile::{tempdir_in, TempDir};
+use tokio::sync::RwLock;
+
+use super::{RepositoryPlugin, RepositoryPluginBuild};
+
+pub const PLUGIN_NAME: &str = "nebula";
+
+const NEBULA_CONFIG_PATH: &str = "/etc/kbs/plugin/nebula-config.toml";
+const CRT_FILENAME: &str = "node.crt";
+const KEY_FILENAME: &str = "node.key";
+
+// Required binaries must be in PATH
+const NEBULA_CERT_BIN: &str = "nebula-cert";
+
+fn create_dirname(path: &Path) -> Result<()> {
+    if let Some(dir) = path.parent() {
+        if !dir.exists() {
+            fs::create_dir_all(dir).with_context(|| format!("Create {} dir", dir.display()))?;
+        }
+    }
+    Ok(())
+}
+
+fn remove_file(path: &Path) -> Result<()> {
```

Comment:
This also seems slightly unnecessary but I guess it doesn't hurt.

Reply:
I removed it as well and moved code around